### ISSUE TITLE
build: deploy workflow has a hard-coded file-name reference to the Guidelines PDF

### ISSUE THREAD

**Issue by bwbohl** [music-encoding/music-encoding#1125]
https://github.com/music-encoding/music-encoding/blob/382f64072f53dc6062283383c1d80c7d9b00dbf1/.github/workflows/deploy.yml#LL107C50-L107C50

**Comment by musicEnfanthen** [comment]
Also, the PDF version should be v5.0.0 OR dev rather than v5.0.0-dev.

**Comment by bwbohl** [comment]
Hmm…, at the moment, it reflects the allowed value of `@meiversion`, but in general, I'm not sure how to deal with this. The `-dev` is there to indicate that it's not a tagged release version, but something under development. Question is though: to what should the version be changed when after a release, additional work is done on the develop branch?
Of course, in terms of harmonizing the version with the folders in the schema repository `dev` (alone) would be the right thing, but in terms of semantic versioning `dev` would be insufficient, does not differentiate between 5.0.0-dev and 6.0.0-dev.
What is clear to me is: the version should be changed with the first change to develop after a release in order to always ell the difference between stable and develop versions of MEI.
What irritates me with the current version number is that it tries to tell what the next version of MEI will be and adds what [semver.org](https://semver.org/) calls a ’prerelease identifier‘ (`-dev`). While the latter is no problem, it could happen that the version on the develop branch is set to the next major release and then a bug is fixed or a new minor change is introduced. This would require a ‘downgrade‘ of the version number: e.g. instead of 6.0.0 we would release 5.1.0 and the version number would have to be changed accordingly.

I'd prefer the reuslt of 
```bash
git describe --tags --always --dirty HEAD
```

This would return something like `v4.0.1-1172-g617bba6a-dirty`. The schematic of which is:

|[NEAREST PAST TAG  NAME]|-|[COMMITS SINCE NEAREST PAST TAG]|-|[SHORT SHA OF LAST COMMIT]|[DIRTY INDICATOR]|
|---|---|---|---|---|---|
|v4.0.1|-|1172|-|g617bba6a|-dirty|

This is an entirely valid semantic version, except for the `v` in the tag name. Moreover it can be created automatically on every state of the clone.

Whatever way we go, any change should also be reflected in the allowed version number in `@meiversion`, but probably as regex pattern.

**Comment by musicEnfanthen** [comment]
Takeaways from discussion at Dev Workshop Charlottesville 2023: 
- reduce version number to 2 levels: 5.0, 5.1, 6.0-dev
- after major release, consider next major release to be dev-version: so after 5.0 new dev version will be 6.0-dev
- publish PDF only on releases, not for all commits

**Comment by bwbohl** [comment]
Meanwhile, I can go with the idea to only have 2-level versioning. Fixing a bug will be transparent for the end-user, especially when it comes to serving the schema from a stable URI. Nevertheless, this might mess with our git tags. After the release of MEI 5.0 (git tag v5.0 or just 5.0) what will the release tag of the first bugfix be?
I'd rather stay with 5.0.0 (both `@meiversion` and git tag) consequently, the above example with the first bugfix release should have 5.0.1 as git tag be published in the schema and guidelines repositories in the 5.0 directory and allow both 5.0.0 and 5.0.1 as values for `@meiversion` and ideally add a schematron rule issuing just a warning that the version should be updated, or even just an info that a bug has been fixed.

**Comment by ahankinson** [comment]
Although I was the first one to propose and implement semantic versioning for MEI, I'm now thinking it's a bad idea, and it's been holding us back.

There are a few realities we have, primarily gained through experience:

1. Our release cycles are very slow, and this is a good thing. We are not building software, we're building a standard.

When we release a version of MEI, it's really just the beginning. People will then need to start encoding their files in MEI. **The specification is not software.** The end of the release cycle is the beginning of the encoding cycle for our community. Changing this too often is bad, since it means that some tool somewhere will not support a newer version, and then our community has to wait for someone to fix this. It also means a lot of churn in our community's projects, since they spend more time upgrading their MEI than they do in writing it.

**Slow is good**, since it gives time for software to catch up, and people to upgrade their existing encodings.

2. It's hard to manage releases, particularly distinguishing between "breaking" vs. "non-breaking" changes. 

We have a single `develop` branch, so all PRs that get approved go there, whether they are major refactorings or minor typo fixes. As we found out, separating the breaking and non-breaking changes _post facto_ is really hard. If we wanted to do "breaking" vs. "non-breaking" we would have two branches for these, and a much more complicated release management. This would be pretty complicated to manage, and we would probably need a dedicated release manager. 

**I think we need to do away with the distinction.** I would even go so far as to do away with the "minor" versions, and just go major-version only. If we figure out that we need to change something after release, we go with non-semver. `v6.patch1` is still v6. `<mei meiversion="6">` would still validate against `.patchN`. Patches are really only if something is actually a regression -- any functional changes, whether they're backwards compatible or not, should wait and go into the next major release.

3. It's hard to communicate with our community about releases.

Our IGs are starting to contribute significantly to the guidelines and the spec, but they have no idea when their stuff will be released. If they start something post v5, then they can aim for v6; they don't have to rush it to get it in to `5.0.1` or something like that. Verovio is also generally targeting the next release in development, so having a simple release structure would be even easier to sync Verovio development with MEI development.

So, in summary

- I'm 100% 👍  for `v5.0` for this release.
- I'm 👎 for continuing with three-digit Semantic Versioning. We gave it a good go, but I really don't think it works for us.
- I would be even more 😃  to even drop the second digit, with the next public release being `v6`, and then `v7`. I really think it will make things a lot simpler.

**Comment by musicEnfanthen** [comment]
Thanks @ahankinson for your thoughts on this. Fully agree that the existing (semver) pattern seems to be an overkill.

In case we decide to use only major versions in the future , I wonder if it wouldn't be better to introduce this change already with v5 (having v4.0.0/4.0.1, v5.0 and then v6 might get odd)?

@bwbohl You can create (non-semver) git tags, right? So v5.0 or v5 should be fine as git release tag. (But maybe I misunderstood your comment?)
